include/gadget/mntns_filter.h

```diff
@@ -3,7 +3,14 @@
 #ifndef MNTNS_FILTER_H
 #define MNTNS_FILTER_H
 
+#ifndef GADGET_TYPE_NETWORKING
+// gadget/mntns.h expects the gadget to include vmlinux.h since it uses struct
+// task_struct, this is an issue for gadgets that don't include vmlinux.h, like
+// networking gadgets. Ideally this file should not be included here, but
+// removing it could break a lot of gadgets, so let's use this hack to allow
+// this working for networking gadgets.
 #include <gadget/mntns.h>
```

Comment:
Could the `#ifndef GADGET_TYPE_NETWORKING` test be moved to `gadget/mntns.h` instead?

Reply:
I could do that, by IMO it's confusing to import a file has all its contents under a  `#ifndef`. My initial thought when implementing this file some time ago was: if anybody is using mntns_filter.h, then they'll need to use mntns.h to get the mount ns, so let's include one in another to avoid the user having to include both of them. But it turns out that on networking gadgets it's not true, as the mount ns is coming from the socket enricher.

Reply:
Ok.

But we don't have a lot of out-of-tree gadgets, so IMO it would be fine as well to update our in-tree gadgets and the documentation to include mntns.h.

Reply:
Handled in the last push.